core/cmd/client.go

```diff
@@ -112,30 +112,43 @@ func (cli *Client) configExitErr(validateFn func() error) cli.ExitCoder {
 	return nil
 }
 
-func (cli *Client) setConfig(opts *chainlink.GeneralConfigOpts, configFiles []string, secretsFile string) error {
+func (cli *Client) initConfigAndLogger(opts *chainlink.GeneralConfigOpts, configFiles []string, secretsFile string) error {
 	if err := loadOpts(opts, configFiles...); err != nil {
 		return err
 	}
 
-	secretsTOML := ""
 	if secretsFile != "" {
 		b, err := os.ReadFile(secretsFile)
 		if err != nil {
 			return errors.Wrapf(err, "failed to read secrets file: %s", secretsFile)
 		}
-		secretsTOML = string(b)
-	}
-	err := opts.ParseSecrets(secretsTOML)
-	if err != nil {
-		return err
+
+		secretsTOML := string(b)
+		err = opts.ParseSecrets(secretsTOML)
+		if err != nil {
+			return err
+		}
 	}
+
 	if cfg, lggr, closeLggr, err := opts.NewAndLogger(); err != nil {
 		return err
 	} else {
+		// If the logger has already been set due to prior initialization, close it out here.
+		if cli.CloseLogger != nil {
+			err := cli.CloseLogger()
+			if err != nil {
+				return errors.Wrap(err, "failed to close initialized logger")
+			}
+		}
+
 		cli.Config = cfg
 		cli.Logger = lggr
 		cli.CloseLogger = closeLggr
 	}
+
+	if len(configFiles) > 0 || secretsFile != "" {
+		cli.configInitialized = true
+	}
```

Comment:
This would allow for some other simplifications too, e.g. unindenting the block behind the `if secretsFile != ""` check on 120.

Reply:
Eh, nvm that breaks the case where they want defaults.

Why move this logic inside of here?

Reply:
@jmank88 Wouldn't that break things? At the moment we can still initialize a config and default logger if we don't provide a config/secrets file. If we're short-circuiting at the top, then anything that requires the default logger will get a nil logger instead.

Reply:
> Eh, nvm that breaks the case where they want defaults.
> 
> Why move this logic inside of here?

A couple of reasons:
* it creates one entry point for initializing a new config and logger
* it encapsulates setting variables like configInitialized

Reply:
Should we rename `configInitialized`? That doesn't seem correct anymore.

Reply:
> it encapsulates setting variables like configInitialized

This is extremely exceptional logic that is only meant to accommodate backwards compatibility until we remove the old way of setting these flags. All else being equal, it would be best to keep it as superficial as possible. There is also an asymmetry now where this field is still checked outside this function from another `Before` - it might be more clear to return a `bool` from this `func`.

Reply:
Yep, can do! With the above, are you referring to just the configInitialized variable, or all the other changes I made to the function as well? (i.e. should I just commit the bit that calls the CloseLogger function?)

Reply:
Just the `configInitilized` var.

Reply:
That looks good. The name is still bugging me though, b/c the config is initialized even if that bool is false. It actually represents whether user input was process - not sure what the best term would be....`flagsProcessed`?